src/constbuffer.c

```diff
@@ -105,26 +105,26 @@ IMPLEMENT_MOCKABLE_FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_CreateFromBuffer,
     /*Codes_SRS_CONSTBUFFER_02_006: [If buffer is NULL then CONSTBUFFER_CreateFromBuffer shall fail and return NULL.]*/
     if (buffer == NULL)
     {
-        LogError("invalid arg passed to CONSTBUFFER_CreateFromBuffer");
+        LogError("invalid arg passed to CONSTBUFFER_CreateFromBuffer buffer: NULL");
         result = NULL;
     }
     else
     {
-        size_t length = BUFFER_length(buffer);
+        uint32_t length = (uint32_t)BUFFER_length(buffer);
```

Comment:
>(uint32_t)BUFFER_length(buffer); [](http://example.com/codeflow?start=26&length=32)

We aren't really using BUFFER_HANDLE's but I'd think if we aren't making that uint32_t as well, then this should at least fail if it's greater than UINT32_MAX

Reply:
Maybe I should just update BUFFER_HANDLE to be uint32_t as well.

Reply:
No, no, don't do that with the same PR please! I went through the same an then you will need to update others too (like base64 and I think string). It just ripples too much, so please contain it for now here so that we get this one step done and we are a bit happier :-).